types/coin.go

```diff
@@ -15,9 +15,13 @@ type Coin struct {
 }
 
 func NewCoin(denom string, amount int64) Coin {
+	return NewIntCoin(denom, NewInt(amount))
+}
+
+func NewIntCoin(denom string, amount Int) Coin {
```

Comment:
I suggest flipping this API. Have NewCoin take in an Int, and have a `NewInt64Coin` method. We want everything to be using sdk.Int normally.